src/Pages/Search/Search.tsx

```diff
@@ -1,35 +1,95 @@
-import { useEffect } from "react";
+// @ts-nocheck
+import { useEffect, useState, useRef } from "react";
+import { LazyLoadImage } from "react-lazy-load-image-component"; //new LazyLoadImage
 import { useAppDispatch, useAppSelector } from "../../app/hook";
 import "./search.css";
 import PokemonCard from "../../components/Card/PokemonCard";
 import { getInitialPokemonData } from "../../app/getInitialData";
 import { getPokemonsData } from "../../app/getPokemonData";
 import InputBar from "../../components/InputBar/InputBar";
-
-const Search:React.FC = () => {
+import Loader from "../../components/Loader/Loader";
+import { increaseOffset } from "../../app/pokemonListSlice";
+const Search: React.FC = () => {
+  // const offsetRef = useRef<number>(0);//redux the offset
+  // const limitRef = useRef<number>(20);// redux the limit
+  const targetRef = useRef<HTMLDivElement>(null);
   const dispatch = useAppDispatch();
-  // data fetching from api and storing it to the store
-   
-  const { allPokemon, randomPokemon } = useAppSelector(
+  const [scrollPosition, setScrollPosition] = useState("");
+  const { allPokemon, randomPokemon, offset, limit } = useAppSelector(
     ({ pokemon }) => pokemon
   );
+
+  const previousOffsetRef = useRef<number>(offset);
   useEffect(() => {
-    dispatch(getInitialPokemonData());
+    previousOffsetRef.current = offset;
+  }, [offset]);
+  const observer = useRef<IntersectionObserver | null>(null);
+
+  useEffect(() => {
+    observer.current = new IntersectionObserver((entries) => {
+      if (entries[0].isIntersecting) {
+        console.log("offSet is " + previousOffsetRef.current);
+        dispatch(getInitialPokemonData(previousOffsetRef.current, limit));
+        dispatch(increaseOffset());
+      }
+    });
+
+    return () => {
+      if (observer.current) {
+        observer.current.disconnect();
+      }
+    };
   }, [dispatch]);
+
+  useEffect(() => {
+    if (targetRef.current && observer.current) {
+      observer.current.observe(targetRef.current);
+    }
+  }, [observer]);
```

Comment:
We don't require a ref for the intersection observer, we can just create a constant.

```suggestion
  useEffect(() => {
   let loadMoreObserver;

   If (targetRef.current) {
     loadMoreObserver = new IntersectionObserver((entries) => {
      if (entries[0].isIntersecting) {
        dispatch(getInitialPokemonData(previousOffsetRef.current, limit));
        }
      });
      
      loadMoreObserver.observe(targetRef.current);
     }

    return () => {
        loadMoreObserver.disconnect();     
    };
  }, [targetRef.current]);
```

Reply:
correct we don't need this ref